view/next-project/src/components/sponsoractivities/EditInvoiceModal.tsx

```diff
@@ -0,0 +1,144 @@
+  };
+
+  return (
+    <Modal className='md:w-1/2'>
+      <div className='w-full'>
+        <div className='ml-auto w-fit'>
+          <CloseButton
+            onClick={() => {
+              setIsOpen(false);
+            }}
+          />
+        </div>
+      </div>
+      <div className='mx-auto mb-10 w-fit text-xl text-black-600'>請求書の修正</div>
+      <div className=''>
+        <div className='my-4 grid grid-cols-5 items-center justify-items-center gap-2'>
+          <p className='text-black-600'>企業名</p>
+          <div className='col-span-4 w-full'>
+            <Input value={editInvoice.sponsorName} onChange={handler('sponsorName')}></Input>
+          </div>
+          <p className='text-black-600'>企業担当者名</p>
+          <div className='col-span-4 w-full'>
+            <Input value={editInvoice.managerName} onChange={handler('managerName')}></Input>
+          </div>
+          <p className='text-black-600'>実行委員担当者名</p>
+          <div className='col-span-4 w-full'>
+            <Input value={editInvoice.fesStuffName} onChange={handler('fesStuffName')}></Input>
+          </div>
+        </div>
+        <div className='max-h-48 overflow-y-auto'>
+          <table className='mb-4 w-full table-fixed border-collapse'>
+            <thead>
+              <tr className='border border-x-white-0 border-b-primary-1 border-t-white-0 py-3'>
+                <th className='w-3/4 px-6 pb-2'>
+                  <div className='text-center text-sm text-black-600'>協賛内容(オプション）</div>
+                </th>
+                <th className='w-1/4 px-6 pb-2'>
+                  <div className='text-center text-sm text-black-600'>値段</div>
+                </th>
+              </tr>
+            </thead>
+            <tbody>
+              {editInvoice.invoiceSponsorStyle &&
+                editInvoice.invoiceSponsorStyle.map((invoiceSponsorStyle, index) => (
+                  <tr
+                    key={index}
+                    className={clsx('border border-x-white-0 border-t-white-0', {
+                      'border-b-primary-1': index === editInvoice.invoiceSponsorStyle.length - 1,
+                    })}
+                  >
+                    <td className='py-3'>
+                      <Input
+                        value={invoiceSponsorStyle.styleName}
+                        className=''
+                        onChange={(e) => {
+                          onChangeSponsorStyle(
+                            { ...invoiceSponsorStyle, styleName: e.target.value },
+                            index,
+                          );
+                        }}
+                      ></Input>
+                    </td>
+                    <td className='py-3'>
+                      <Input
+                        value={invoiceSponsorStyle.price}
+                        onChange={(e) => {
+                          onChangeSponsorStyle(
+                            { ...invoiceSponsorStyle, price: Number(e.target.value) },
```

Comment:
値段に誤って全角文字を入力すると、NaN表示になり修正が不可能になる問題があります。
```suggestion
                            {
                              ...invoiceSponsorStyle,
                              price: isNaN(Number(e.target.value))
                                ? invoiceSponsorStyle.price
                                : Number(e.target.value),
                            },
```

Reply:
ありがとうございます。
[全角の対応](https://github.com/NUTFes/FinanSu/pull/869/commits/0327c835f4ee44c13379189a9bf6933db467fd99)